### ISSUE TITLE
MainActivity; - skipping class java.lang.ArrayIndexOutOfBoundsException: 7

### ISSUE THREAD

**Issue by JesusFreke** [JesusFreke/smali#406]
### Stacktrace/Logcat

```
Error occurred while disassembling class Lcom.some.app.MainActivity; - skipping class
java.lang.ArrayIndexOutOfBoundsException: 7
\tat org.jf.dexlib2.dexbacked.util.DebugInfo$DebugInfoImpl$3.readNextItem(DebugInfo.java:200)
\tat org.jf.dexlib2.dexbacked.util.DebugInfo$DebugInfoImpl$3.readNextItem(DebugInfo.java:157)
\tat org.jf.dexlib2.dexbacked.util.VariableSizeLookaheadIterator.computeNext(VariableSizeLookaheadIterator.java:60)
\tat com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:143)
\tat com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:138)
\tat org.jf.baksmali.Adaptors.MethodDefinition.addDebugInfo(MethodDefinition.java:576)
\tat org.jf.baksmali.Adaptors.MethodDefinition.getMethodItems(MethodDefinition.java:378)
\tat org.jf.baksmali.Adaptors.MethodDefinition.writeTo(MethodDefinition.java:238)
\tat org.jf.baksmali.Adaptors.ClassDefinition.writeDirectMethods(ClassDefinition.java:282)
\tat org.jf.baksmali.Adaptors.ClassDefinition.writeTo(ClassDefinition.java:112)
\tat org.jf.baksmali.baksmali.disassembleClass(baksmali.java:225)
\tat org.jf.baksmali.baksmali.access$000(baksmali.java:55)
\tat org.jf.baksmali.baksmali$1.call(baksmali.java:149)
\tat org.jf.baksmali.baksmali$1.call(baksmali.java:147)
\tat java.util.concurrent.FutureTask.run(FutureTask.java:262)
\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
\tat java.lang.Thread.run(Thread.java:745)
```

### Steps to Reproduce
1. Add line: \".end local vX\" (X is a number) to onCreate() method of some activity.smali that you dont want to be decompiled.
2. Recompile it and sign it.
3. Apk can be installed on your Android smartphone without problem, but can't be decompiled and recompiled with apktool.

### More info
Some modders are using this trick to avoid people decompiling their apps. Some mod-apps that include this trick are YoWhatsapp and GBWhatsapp 

(copy of apktool issue: https://github.com/iBotPeaches/Apktool/issues/1269)

**Comment by JesusFreke** [comment]
More details from IRC:

    <Aex12> vX should be a number higher than .locals Y to make a ArrayIndexOutOfBounds exception
    <Aex12> YoWhatsapp is using .end local v77 to make their code undecompilable

**Comment by Aex12** [comment]
I found a temporary solution for this error. Now I can decompile any APK with this protection.
In `org.jf.dexlib2.dexbacked.util.DebugInfo.java` I added these lines before line 200:
```
if (register >= registerCount) {
    System.out.println("Found \".end local v"+register+"\" in \".locals v"+registerCount+"\" method. Skipping corrupted line.");
    continue;
}
```
**Code in context:**
```
case DebugItemType.END_LOCAL: {
    int register = reader.readSmallUleb128();
    if (register >= registerCount) {
    	System.out.println("Found \".end local v"+register+"\" in \".locals v"+registerCount+"\" method. Skipping corrupted line.");
    	continue;
    }
    LocalInfo localInfo = locals[register];
    boolean replaceLocalInTable = true;
    if (localInfo instanceof EndLocal) {
        localInfo = EMPTY_LOCAL_INFO;
        // don't replace the local info in locals. The new EndLocal won't have any info at all,
        // and we dont want to wipe out what's there, so that it is available for a subsequent
        // RestartLocal
        replaceLocalInTable = false;
    }
    ImmutableEndLocal endLocal =
            new ImmutableEndLocal(codeAddress, register, localInfo.getName(),
                    localInfo.getType(), localInfo.getSignature());
    if (replaceLocalInTable) {
        locals[register] = endLocal;
    }
    return endLocal;
}
```

**Comment by Aex12** [comment]
Any progress on that?

**Comment by JesusFreke** [comment]
Fixed in c347e68b
